### PR TITLE
Remove unused feature gates

### PR DESCRIPTION
**Pull request by ashleyschuett**
**What this PR does / why we need it**:

Removing unused feature gates from the kubevirt configuration. 

Since these feature gate flags are not checked I think they should be removed. 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-config/feature-gates.go [hunk 1]
```diff
@@ -32,8 +32,6 @@ const (
 	SidecarGate           = "Sidecar"
 	GPUGate               = "GPU"
 	SnapshotGate          = "Snapshot"
-	DataVolumesGate       = "DataVolumes"
-	SRIOVGate             = "SRIOVGate"
```
Comment (rmohr) [inline pkg/virt-config/feature-gates.go:36]: @vladikr I am surpised that SRIOV is not gated. I thought that it involves some extra privileges.

Comment (vladikr) [reply pkg/virt-config/feature-gates.go:36]: @rmohr seems like you've approved it: https://github.com/kubevirt/kubevirt/pull/2546

Comment (vladikr) [reply pkg/virt-config/feature-gates.go:36]: @rmohr, afaik, there are no additional privileges with sr-iov, but we will lock 1G of memory for VMIs with SR-IOV.

Comment (rmohr) [reply pkg/virt-config/feature-gates.go:36]: > @rmohr seems like you've approved it: #2546

ups

> @rmohr, afaik, there are no additional privileges with sr-iov, but we will lock 1G of memory for VMIs with SR-IOV.

 perfect that was what I was unsure about.

**Pull request merged**